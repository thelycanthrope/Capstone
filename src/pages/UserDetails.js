import React, { useEffect, useState } from 'react';
import { useSelector,useDispatch } from 'react-redux';
import { fetchUser} from '../Actions/actions'
import NavBar from '../Components/NavBar';
import './page.css'


let user={

  email: "",
  password: "",
  fname: "",
  lname:"" ,
  location: "",
  contactno:""

}

function set  (users,email)  {

  for( let i=0;i<users.users.length;i++){
    if(email===users.users[i].email){
      user={...users.users[i]}
      console.log(user);
    }
  }
  
}


const UserDetails = () => {
  const [email,setEmail]=useState()
  

const users=useSelector( state => state )

  const dispatch = useDispatch()
  
  useEffect ( ( ) =>{
    dispatch(fetchUser()) },[dispatch])


  useEffect ( ( ) =>{ 
//     axios.get('http://localhost:5500/loggedin/1')
// .then ( (r) => {setEmail(r.data.email)})
// .catch( err => console.log(err)) 
setEmail(localStorage.getItem('loggedin'))
},[])


console.log(email)


email && users ? set(users,email):console.log('else')



return (
  <div className="bg-dark ">
    <NavBar/>
    
    <br></br>
    <br></br>
    <br></br>
    <br></br>
    <br></br>
    <br></br>
    <br></br>
    <br></br>
    <br></br>
    <div className="container text-white d-flex justify-content-center ">
      <div className="card text-center bg-dark border" style={{ width: "25rem" }}>
        <div className="card-body">
          <h5 className="card-title" >Welcome, <span className="bg-success ">&nbsp;{user.fname} {user.lname}&nbsp;</span></h5>
          <p className="card-text">Registerd Email: <span className="bg-warning text-dark" style={{fontWeight:"bold"}}>&nbsp;{user.email}&nbsp;</span></p>
          <p className="card-text ">Registered Contact No: <span className="bg-primary" style={{fontWeight:"bold"}}>&nbsp;{user.contactno}&nbsp;</span></p>
          <p className="card-text">Registered Location: <span className="bg-light text-danger" style={{fontWeight:"bold"}}>&nbsp;{user.location}&nbsp;</span></p>
          <div className="card-footer text-muted border">

          </div>


        </div>
      </div>
    </div>
    <br></br>
    <br></br>
    <br></br>
    <br></br>
    <br></br>
    <br></br>
    <br></br>
    <br></br> 
  </div>
  );
};

export default UserDetails;
