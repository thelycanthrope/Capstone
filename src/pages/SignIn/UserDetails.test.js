import React from 'react'
import {shallow,configure,mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import UserDetails from '../UserDetails';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';


configure({ adapter: new Adapter() });
const mockStore = configureStore([]);

describe('snapshot', () => {
    let store;
    let wrapper;
    beforeEach(() => {
        store = mockStore({
            myState: 'sample text',
        })
        wrapper = shallow(<Provider store={store}><UserDetails /></Provider>)
    })


    it('snapshot test',() =>{
        const wrapper=shallow(<BrowserRouter><Provider store={store}><UserDetails /></Provider></BrowserRouter>).dive().render()
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('should have 1 h5',() =>{
        const wrapper=shallow(<BrowserRouter><Provider store={store}><UserDetails /></Provider></BrowserRouter>).dive().render()
        expect(wrapper.find('h5').length).toEqual(1)
    })
    it('should have  3 p tag',() =>{
        const wrapper=shallow(<BrowserRouter><Provider store={store}><UserDetails /></Provider></BrowserRouter>).dive().render()
        expect(wrapper.find('p').length).toEqual(3)
    })
})

