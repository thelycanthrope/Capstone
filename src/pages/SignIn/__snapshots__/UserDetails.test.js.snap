// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot snapshot test 1`] = `
<div
  class="bg-dark "
>
  <br />
  <br />
  <br />
  <br />
  <br />
  <br />
  <br />
  <br />
  <br />
  <div
    class="container text-white d-flex justify-content-center "
  >
    <div
      class="card text-center bg-dark border"
      style="width:25rem"
    >
      <div
        class="card-body"
      >
        <h5
          class="card-title"
        >
          Welcome, 
          <span
            class="bg-success "
          >
               
          </span>
        </h5>
        <p
          class="card-text"
        >
          Registerd Email: 
          <span
            class="bg-warning text-dark"
            style="font-weight:bold"
          >
              
          </span>
        </p>
        <p
          class="card-text "
        >
          Registered Contact No: 
          <span
            class="bg-primary"
            style="font-weight:bold"
          >
              
          </span>
        </p>
        <p
          class="card-text"
        >
          Registered Location: 
          <span
            class="bg-light text-danger"
            style="font-weight:bold"
          >
              
          </span>
        </p>
        <div
          class="card-footer text-muted border"
        />
      </div>
    </div>
  </div>
  <br />
  <br />
  <br />
  <br />
  <br />
  <br />
  <br />
  <br />
</div>
`;
